runbotics-orchestrator/src/main/java/com/runbotics/service/impl/ProcessServiceImpl.java

```diff
@@ -59,18 +62,26 @@ public ProcessServiceImpl(
         this.userService = userService;
         this.botCollectionService = botCollectionService;
         this.globalVariableService = globalVariableService;
+        this.processCollectionService = processCollectionService;
     }
 
     @Override
     public ProcessDTO save(ProcessDTO processDTO) {
         log.debug("Request to save Process : {}", processDTO);
+        User user = userService.getUserWithAuthorities().get();
+
         Process process = processMapper.toEntity(processDTO);
         process.setUpdated(ZonedDateTime.now());
         if (process.getId() == null) {
-            User user = userService.getUserWithAuthorities().get();
             process.setCreated(ZonedDateTime.now());
             process.setCreatedBy(user);
             process.setEditor(user);
+        } else {
+            Process existingProcess = processRepository.findById(
+                processDTO.getId()).orElseThrow(() -> new BadRequestAlertException("Cannot find process with this id", ENTITY_NAME, "processNotFound")
+            );
```

Comment:
for code readability maybe it is better to assign it to `processId` variable?

Reply:
You're right. I will do it :)